merkledag/node.go

```diff
@@ -131,7 +133,7 @@ func (n *ProtoNode) AddRawLink(name string, l *ipld.Link) error {
 
 // RemoveNodeLink removes a link on this node by the given name.
 func (n *ProtoNode) RemoveNodeLink(name string) error {
-	n.encoded = nil
+	n.invalidateEncoding()
```

Comment:
Nit: we're not invalidating the "encoding", just the "encoded" or cached object (I'd call it `invalidateCache`.